src/utils/project_folder.rs

```diff
@@ -0,0 +1,46 @@
+use std::{env, path::PathBuf};
+use std::error::Error;
+use std::fmt;
+use std::fs::create_dir;
+use std::result::Result;
+
+use dirs::home_dir;
+use sha2::Digest;
+
+#[derive(Debug)]
+struct ProjectFolderError(String);
```

Comment:
Can you rename this to `ProjectDirectoryError`?

Reply:
Renamed